### PR TITLE
Exp pr

### PR DESCRIPTION
**Pull request by liususan091219**
exp -> main

### CODE REVIEW

File: flaml/nlp/autotransformers.py [hunk 1]
```diff
@@ -48,7 +52,7 @@ def _convert_dict_to_ray_tune_space(config_json, mode="grid"):
             for each_hp in config_json.keys():
                 this_config = config_json[each_hp]
                 assert isinstance(this_config, dict) or isinstance(this_config, list), \
-                    "config of " + each_hp + " must be dict or list for grid search"
+                    "config of " + each_hp + " must be dict or list"
```
Comment (sonichi) [inline flaml/nlp/autotransformers.py:55]: recover

File: flaml/nlp/autotransformers.py [hunk 2]
```diff
@@ -74,9 +78,14 @@ def _set_search_space(self,
                           **custom_hpo_args):
         from .hpo.hpo_searchspace import AutoHPOSearchSpace
 
+        if self.jobid_config.spa == "grid":
+            model_type = "bert"
```
Comment (sonichi) [inline flaml/nlp/autotransformers.py:82]: Does 'grid' only work for bert?
Please rename `.spa` to more informative name

File: flaml/nlp/autotransformers.py [hunk 3]
```diff
@@ -109,38 +177,43 @@ def prepare_data(self,
...
                      resplit_portion=None,
+                     load_config_mode="console",
                      **custom_data_args):
-        """Prepare data
-
-            Example:
+        '''Prepare data
 
-                .. code-block:: python
+            An example:
```
Comment (sonichi) [inline flaml/nlp/autotransformers.py:184]: Why change?

File: flaml/nlp/autotransformers.py [hunk 4]
```diff
@@ -169,73 +244,102 @@ def prepare_data(self,
...
-        if self.jobid_config.spt == "rspt":
+        if self.jobid_config.spt.endswith("rspt"):
             assert resplit_portion, "If split mode is 'rspt', the resplit_portion must be provided. Please " \
                                     "refer to the example in the documentation of AutoTransformers.prepare_data()"
         if self.jobid_config.subdat:
             data_raw = load_dataset(JobID.dataset_list_to_str(self.jobid_config.dat),
                                     self.jobid_config.subdat)
         else:
-            data_raw = load_dataset(*self.jobid_config.dat)
+            data_raw = AutoTransformers._wrapper(load_dataset, *self.jobid_config.dat)
```
Comment (sonichi) [inline flaml/nlp/autotransformers.py:254]: I removed `_wrapper`

File: flaml/nlp/autotransformers.py [hunk 5]
```diff
@@ -169,73 +244,102 @@ def prepare_data(self,
...
+                        self.eval_dataset = this_encoded_data
+                    else:
+                        self.test_dataset = this_encoded_data
+                    _max_seq_length = max(_max_seq_length,
+                                               max([sum(this_encoded_data[x]['attention_mask']) for x in
+                                                    range(len(this_encoded_data))]))
+                self._max_seq_length = int((_max_seq_length + 15) / 16) * 16
+            else:
+                assert "foldnum" in custom_data_args, "if the split mode is cross validation, foldnum must be" \
+                                                      "specified"
```
Comment (sonichi) [inline flaml/nlp/autotransformers.py:306]:
```suggestion
                                                      " specified"
```

File: flaml/nlp/autotransformers.py [hunk 6]
```diff
@@ -433,10 +539,12 @@ def _verify_init_config(self,
         for key in custom_hpo_args.keys():
             if key == "points_to_evaluate":
                 for each_init_config in custom_hpo_args[key]:
-                    for each_hp in each_init_config.keys():
-                        assert each_hp in self._search_space_hpo.keys(), \
-                            "points_to_evaluate hp must be within the search space"
+                    for each_hp in list(each_init_config.keys()):
+                        if each_hp not in self._search_space_hpo.keys():
+                            del each_init_config[each_hp]
+                        print("{} is not in the search space, deleting from init config".format(each_hp))
```
Comment (sonichi) [inline flaml/nlp/autotransformers.py:545]: why do we allow keys not in the search space?

**Review by sonichi — COMMENTED** [review]
Many of my changes earlier were removed. Please cover. Also, please follow  https://github.com/microsoft/FLAML#pre-commit.

File: flaml/nlp/autotransformers.py [hunk 7]
```diff
@@ -29,11 +32,12 @@ class AutoTransformers:
...
-                "resources_per_trial": {"cpu": 1, "gpu": 1},
-                "num_samples": -1,
-                "time_budget": 60,
-            }
+            autohf_settings = {"resources_per_trial": {"cpu": 1},
+                       "num_samples": -1,
+                       "time_budget": 100000,
+                       "ckpt_per_epoch": 1,
+                       "fp16": False,
+                      }
```
[the remark this answers is not in the document]
Comment (liususan091219) [reply flaml/nlp/autotransformers.py:40]: automatically recovered after fetching upstream

File: flaml/nlp/autotransformers.py (hunk 1 above)
Comment (liususan091219) [reply flaml/nlp/autotransformers.py:55]: automatically recovered after fetching upstream

File: flaml/nlp/autotransformers.py (hunk 6 above)
Comment (liususan091219) [reply flaml/nlp/autotransformers.py:545]: Not allowing, we are deleting it.

File: flaml/nlp/autotransformers.py (hunk 2 above)
Comment (liususan091219) [reply flaml/nlp/autotransformers.py:82]: "Does 'grid' only work for bert?" => fixed. 

"Please rename .spa to more informative name" => spa is the abbreviated name for space_mode. The abbreviated name is only for logging in azure, not meant to be informative. The informative version for user is defined in nlp/utils.py

Comment (sonichi) [reply flaml/nlp/autotransformers.py:82]: not informative for developers either. rename for code readability and maintainability.

File: flaml/nlp/autotransformers.py (hunk 6 above)
Comment (sonichi) [reply flaml/nlp/autotransformers.py:545]: use logger.warning?

**Pull request closed without merging**